### ISSUE TITLE
piglow: presence of both Shutdown and Close confuses

### ISSUE THREAD

**Issue by rakyll** [goiot/devices#25]
Isn't it possible to do the shutdown in Close? Is there a reason why Shutdown has to be a separate method?

**Comment by mattetti** [comment]
I was wondering the same. Is there value in shutting down without closing?

**Comment by zankich** [comment]
`Shutdown` puts the PiGlow into a "software shutdown" mode, but it does not sever the connection to the PiGlow. Most of the methods in the PiGlow package are mapped directly to their register functions according to the api specification.

**Comment by mattetti** [comment]
@zankich can you think of a good reason why someone would want to shutdown and not close the device? If you do, we should probably keep both methods, but either way, I think that we should always shutdown during closing tho, what do you think?

**Comment by zankich** [comment]
@mattetti I can imagine a case where you want to set a persistent color on your PiGlow, but do not want to have a program running in the background. Some process may wake up and set the colors to a specific pattern and then exit, or if you have a more complex system a routine which opens a connection to the PiGlow and and then closes the connection when it's finished. If you sent the PiGlow a `Shutdown`, the lights would turn off.

**Comment by mattetti** [comment]
Hmm that's very interesting. That's a totally realistic use case IMHO and does justify the 2 methods. Maybe some documentation might clarify why there are two methods. After all both Burcu and I were surprised.

**Comment by zankich** [comment]
Yeah the documentation could be clearer for sure. I took a first stab at it with my PR, but there is definitely room to improve.

**Comment by rakyll** [comment]
>  I can imagine a case where you want to set a persistent color on your PiGlow, but do not want to have a program running in the background.

You can close the file descriptor and it will keep using displaying the latest state. These devices are state machines, not going to stop working when you close the I2C connection through devfs.
